### PR TITLE
ODC-7596: Show Pod Disruption warning message in Topology page

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Story:**
https://issues.redhat.com/browse/ODC-7596

**Description:**
If there is any PodDisruptionBudget which is violated, added a warning message in Topology page to let the user know about this violation.

**Screenshots/Video:**

https://github.com/openshift/console/assets/102503482/6950c7dd-b30c-4945-bfc7-9eba80640860

Screenshot 2024-06-03 at 11 12 51 AM
Screenshot 2024-06-03 at 11 36 59 AM
Screenshot 2024-06-03 at 11 33 09 AM
Screenshot 2024-06-03 at 11 32 44 AM

--------------

**Unit tests:**
Screenshot 2024-06-06 at 12 41 19 PM

--------------

**e2e tests:**
Screenshot 2024-06-06 at 12 40 56 PM

### CODE REVIEW

File: frontend/packages/console-app/src/components/pdb/utils/get-pdb-resources.ts [hunk 1]
```diff
@@ -10,3 +15,31 @@ export const getPDBResource = (
...
+  const conditions = status?.conditions;
+
+  const disruptionNotAllowedCondition = conditions?.find(
+    (condition: PDBCondition) =>
+      condition.type === 'DisruptionAllowed' &&
+      condition.status === K8sResourceConditionStatus.False,
+  );
+  return !!disruptionNotAllowedCondition;
+};
+
```
Comment (cyril-ui-developer) [inline frontend/packages/console-app/src/components/pdb/utils/get-pdb-resources.ts:30]: The condition for checking for PDB violation(s) should consider the pod label selector in the applied pod controller(s). If there is a PDB which does not match any pod label selector in the specified namespace, the warning alert is shown. I think that is incorrect. See the screenshots for example


Screenshot 2024-06-04 at 2 15 50 PM

Screenshot 2024-06-04 at 2 16 22 PM

Screenshot 2024-06-04 at 2 55 41 PM

The PDB object `status.expectedPods` will be `0` if it doesn’t match any pod controllers label selector. I think we should include that in the condition.

Comment (lokanandaprabhu) [reply frontend/packages/console-app/src/components/pdb/utils/get-pdb-resources.ts:30]: Updated the PR. PTAL.

**Review by vikram-raj — COMMENTED** [review]
Verified it. 
/lgtm

**Comment by sanketpathak** [comment]
Verified on a cluster with admin and non-admin users. Tests are running green as well


https://github.com/openshift/console/assets/10252230/c05b749c-1151-47e8-b301-c801330e1368


Screenshot 2024-06-13 at 8 06 32 PM
/label qe-approved

**Pull request merged**